src/sentry/taskworker/taskworker.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+import logging
+import time
+from datetime import datetime
+from multiprocessing.context import TimeoutError
+from multiprocessing.pool import Pool
+from uuid import uuid4
+
+import grpc
+import orjson
+from django.conf import settings
+from sentry_protos.sentry.v1.taskworker_pb2 import (
+    TASK_ACTIVATION_STATUS_COMPLETE,
+    TASK_ACTIVATION_STATUS_FAILURE,
+    TASK_ACTIVATION_STATUS_RETRY,
+    TaskActivation,
+)
+
+from sentry.taskworker import taskworker_process
+from sentry.taskworker.registry import TaskNamespace, taskregistry
+
+logger = logging.getLogger("sentry.taskworker")
+result_logger = logging.getLogger("taskworker.results")
+
+
+class WorkerCompleteException(Exception):
+    pass
+
+
+class TaskWorker:
+    """
+    A TaskWorker fetches tasks from a task queue (inflight actiivation store) and handles executing them.
+    As a tasks are executed, the worker will update the task status in the task queue.
+    """
+
+    def __init__(self, **options):
+        self.options = options
+        self.exitcode = None
+        self._execution_count = 0
+        self._worker_id = uuid4().hex
+        self._namespace: TaskNamespace | None = None
+        self._build_pool()
```

Comment:
We might want to defer the pool construction until `start()`. That might help with django setup errors too.